### PR TITLE
Fix s3 API not being compatible with aws command line

### PR DESCRIPTION
**Pull request by bradyoo**
Both S3Browser and AWS command line are particular about the output of these files in different ways. Fortunately, s3 API is very visible about it and can tell what it's doing. The cli on the other hand will just throw cryptic errors at you until you trial & error it out.

The two important changes here are:

1. each entry of CommonPrefixes has its own <CommonPrefixes>
2. Add a "/" at the end of each common prefix.

### CODE REVIEW

**Comment by bradyoo** [comment]
Normally Gene would review this but given Gene may not be able to get to it, giving to another person as well (Shouwei). It could have been any random engineer since nobody else has seen this code.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java [hunk 1]
```diff
@@ -110,13 +110,14 @@ public ListBucketResult(
       ));
     }
 
-    final ArrayList<String> commonPrefixes = new ArrayList<>();
+    final ArrayList<Prefix> commonPrefixes = new ArrayList<>();
     for (URIStatus status : prefixList) {
       final String path = status.getPath();
-      commonPrefixes.add(path.substring(mName.length() + 2)); // remove both ends of "/" character
+      // remove both ends of "/" character, add "/" at end to show it's prefix
+      commonPrefixes.add(new Prefix(path.substring(mName.length() + 2) + "/"));
```
[the remark this answers is not in the document]
Comment (bradyoo) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java:117]: aws cli expects a "/" at the end for folders or else the cli hard crashes with index out of bounds exception. 

It's hard to tell what the exact thing that the cli wants unless I inspect the cli code directly.

**Comment by bradyoo** [comment]
Added comment

**Review by HelloHorizon — APPROVED** [review]
LGTM after build pass

**Comment by bradyoo** [comment]
alluxio-bot, merge this please.

**Comment by bradyoo** [comment]
alluxio-bot, cherry-pick this to branch-2.5 please.

**Pull request merged**